src/ts/api.ts

```diff
@@ -161,20 +163,36 @@ async function estimateTradeAmount({
   amount,
   baseUrl,
 }: EstimateTradeAmountQuery & ApiCall): Promise<BigNumber> {
-  const response: EstimateAmountResponse = await call(
-    `markets/${sellToken}-${buyToken}/${apiKind(kind)}/${BigNumber.from(
-      amount,
-    ).toString()}`,
-    baseUrl,
+  const side =
+    kind == OrderKind.SELL
+      ? ({
+          kind: OrderKind.SELL,
+          sellAmountAfterFee: amount,
+        } as SellAmountAfterFee)
+      : ({
+          kind: OrderKind.BUY,
+          buyAmountAfterFee: amount,
+        } as BuyAmountAfterFee);
```

Comment:
We could use ` const side: BuyAmountAfterFee | SellAmountAfterFee =` instead of type castings to make sure the expression is still properly typed.

Reply:
Fixed in 4c3c796e3df713587411f27aa95966fde3166fbb